src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/debtposition/InstallmentProcessingService.java

```diff
@@ -0,0 +1,106 @@
+import it.gov.pagopa.payhub.activities.dto.ingestion.debtposition.InstallmentIngestionFlowFileDTO;
+import it.gov.pagopa.payhub.activities.dto.ingestion.debtposition.InstallmentIngestionFlowFileResult;
+import it.gov.pagopa.payhub.activities.mapper.ingestionflow.debtposition.InstallmentSynchronizeMapper;
+import it.gov.pagopa.pu.debtposition.dto.generated.InstallmentSynchronizeDTO;
+import it.gov.pagopa.pu.processexecutions.dto.generated.IngestionFlowFile;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
+
+@Service
+@Lazy
+@Slf4j
+public class InstallmentProcessingService {
+
+    private final DebtPositionService debtPositionService;
+    private final InstallmentSynchronizeMapper installmentSynchronizeMapper;
+    private final InstallmentErrorsArchiverService installmentErrorsArchiverService;
+    private final WorkflowCompletionService workflowCompletionService;
+
+    public InstallmentProcessingService(DebtPositionService debtPositionService,
+                                        InstallmentSynchronizeMapper installmentSynchronizeMapper,
+                                        InstallmentErrorsArchiverService installmentErrorsArchiverService,
+                                        WorkflowCompletionService workflowCompletionService) {
+        this.debtPositionService = debtPositionService;
+        this.installmentSynchronizeMapper = installmentSynchronizeMapper;
+        this.installmentErrorsArchiverService = installmentErrorsArchiverService;
+        this.workflowCompletionService = workflowCompletionService;
+    }
+
+    /**
+     * Processes a stream of InstallmentIngestionFlowFileDTO and synchronizes each installment.
+     *
+     * @param installmentIngestionFlowFileDTOStream Stream of installment ingestion flow file DTOs to be processed.
+     * @param ingestionFlowFile Metadata of the ingestion file containing details about the ingestion process.
+     * @param workingDirectory The directory where error files will be written if processing fails.
+     * @return An {@link InstallmentIngestionFlowFileResult} containing details about the processed rows, errors, and archived files.
+     */
+    public InstallmentIngestionFlowFileResult processInstallments(Stream<InstallmentIngestionFlowFileDTO> installmentIngestionFlowFileDTOStream,
+                                                                  IngestionFlowFile ingestionFlowFile,
+                                                                  Path workingDirectory) {
+        List<InstallmentErrorDTO> errorList = new ArrayList<>();
+        AtomicLong processedRows = new AtomicLong(0);
+
+        long totalRows = installmentIngestionFlowFileDTOStream
```

Comment:
this is not totalRows if you are applying a filter... this is the number of elements which have passed the filter.
You could use the filter to count just the correctly processedRows, while inside the filter, as first thing, you could increment the totalRows, handled as an explicit counter (instead of processedRows)